### PR TITLE
feat: adobe fonts provider

### LINKED ISSUE #11: adobe fonts provider

### PR DESCRIPTION
**Pull request by qwerzl**
This PR resolves #11. What is different from the proposal written in the issue is that when a font is matched in the kit, a regex filters out the `@font-face` declarations that are not related to the font requested.

I think the next thing we can work on this provider is to work with different kit IDs and treat them like separate providers.

### CODE REVIEW

File: src/providers/adobe.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
+import { $fetch } from 'ofetch'
+import { hash } from 'ohash'
+
+import type { FontProvider, ResolveFontFacesOptions } from '../types'
+import { extractFontFaceData, addLocalFallbacks } from '../css/parse'
+import { cachedData } from '../cache'
+import { logger } from '../logger'
+
+interface ProviderOption {
+  id?: string
```
Comment (danielroe) [inline src/providers/adobe.ts:10]: in terms of your proposed next steps, should we perhaps also support an array of ids?

File: playground/nuxt.config.ts [hunk 2]
```diff
@@ -16,7 +16,11 @@ export default defineNuxtConfig({
       { name: 'MyCustom', src: '/custom-font.woff2' },
       { name: 'CustomGlobal', global: true, src: '/font-global.woff2' },
       { name: 'Oswald', fallbacks: ['Times New Roman'] },
+      { name: 'Aleo', provider: 'adobe'}
     ],
+    adobe: {
+      id: "sij5ufr",
```
Comment (danielroe) [inline playground/nuxt.config.ts:22]:
```suggestion
      id: 'sij5ufr',
```

File: src/providers/adobe.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  const font = fonts.kit.families.find(f => f.name === family)!
+  const styles: string[] = []
+  for (const style of font.variations) {
+    if (style.includes('i') && !variants.styles.includes('italic')) { continue }
+    if (!variants.weights.includes(String(style.slice(-1)+'00'))) { continue }
+    styles.push(style)
+  }
+
+  if (styles.length === 0) return []
+  const fontFaceRegex = new RegExp(`@font-face\\s*{[^}]*font-family:\\s*["']${family.toLowerCase()}["'][^}]*}`, 'g');
```
Comment (danielroe) [inline src/providers/adobe.ts:87]: possibly we could also pass a font family to `extractFontFaceData` and handle this in the parsing instead ...

File: src/providers/adobe.ts [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+}
+
+let fonts: AdobeFontMeta
+const familyMap = new Map<string, string>()
+
+async function initialiseFontMeta (kitId: string) {
+  fonts = await cachedData('adobe:meta.json', () => fontAPI<AdobeFontMeta>(`/api/v1/json/kits/${kitId}/published`, { responseType: 'json' }), {
+    onError () {
+      logger.error('Could not download `adobe` font metadata. `@nuxt/fonts` will not be able to inject `@font-face` rules for adobe.')
+      return {kit: { id: '', families: [] } }
```
Comment (danielroe) [inline src/providers/adobe.ts:64]:
```suggestion
      return { kit: { id: '', families: [] } }
```

**Review by danielroe — COMMENTED** [review]
You're so fast! This looks great. ❤️

File: src/providers/adobe.ts [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+import { cachedData } from '../cache'
+import { logger } from '../logger'
+
+interface ProviderOption {
+  id?: string
+}
+
+export default {
+  async setup (providerOption: ProviderOption) {
+    await initialiseFontMeta(providerOption.id!)
```
Comment (danielroe) [inline src/providers/adobe.ts:15]: I think we should early return if there's no id or array of ids.

**Comment by qwerzl** [comment]
All resolved. Also fixed the bug where font faces with spaces in their names can't get matched.

**Review by danielroe — APPROVED** [review]
Thank you so much! ❤️

**Pull request merged**